src/WebJobs.Extensions.DurableTask/DurableOrchestrationContext.cs

```diff
@@ -430,29 +430,29 @@ internal void RaiseEvent(string name, string input)
         {
             lock (this.pendingExternalEvents)
             {
-                object tcsObject;
-                if (this.pendingExternalEvents.TryGetValue(name, out tcsObject))
+                ICollection tcsColleciton;
+                if (this.pendingExternalEvents.TryGetValue(name, out tcsColleciton))
                 {
-                    var tcsList = tcsObject as IList;
-                    if (tcsList == null || tcsList.Count == 0)
+                    if (tcsColleciton == null || tcsColleciton.Count == 0)
                     {
                         return;
                     }
 
-                    var tcs = tcsList[0];
+                    Queue tcsQueue = new Queue(tcsColleciton);

```

Comment:
The downside of this approach is that the contents of the queue is being copied every time we receive an event. This is very inefficient, and potentially worse than the `IList.RemoveAt(0)` performance from the previous iteration.

Reply:
If we use `Queue` instead of `ICollection`, then this inefficient conversion won't be necessary.

Reply:
Great! I will make the modifications. Thank you!